### ISSUE TITLE
[Internship Project]: Riley Pittman

### ISSUE THREAD

**Issue by GemmaTuron** [ersilia-os/ersilia#715]
### Summary

Hello,

This is a public issue for a virtual daily stand-up. We will use this to briefly share the tasks of the day and the challenges and advances made, so that we can ensure smooth support from the Ersilia mentors and alignment between daily tasks and overall internship goals.

### Scope

Initiative 🐋

### Objective(s)

Internship goals:

- Learn how to work in an Open Source Community
- Improve the infrastructure of the Ersilia Model Hub
- Improve the usability and platform compatibility of Ersilia Models, mainly via Docker Images
- Learn about AI/ML models for drug discovery
- Identify new relevant AI/ML models and datasets and add them to this [discussion](https://github.com/ersilia-os/ersilia/discussions/694)
- Incorporate new AI/ML models in the Hub
- Improve the documentation of the Ersilia Model Hub for users
- Improve the documentation of the Ersilia Model Hub for developers

### Team

| Role & Responsibility | Username(s) | 
|--|--|
| Intern | @pittmanriley |
| Mentor| @miquelduranfrigola  |
| Coordinator| @GemmaTuron  |

### Timeline

Before starting your work, line up a few tasks and short description. This should not take long. For example, it could be something like:
Wednesday 21st June
- [ ] Create Intern daily task tracker
- [ ] Update the Ersilia GitHub Project
- [ ] Make sure Git Actions start upon success of the other workflows
- [ ] Work on the documentation for model testing
- [ ] Attend the Drug Discovery intro by WCAIR

**Comment by GemmaTuron** [comment]
In this example, following the tasks of Wednesday 21st June:

- [x]  Create Intern daily task tracker
- [x]  Update the Ersilia GitHub Project
- [x]  Make sure Git Actions start upon success of the other workflows
- [ ]  Work on the documentation for model testing
- [x]  Attend the Drug Discovery intro by WCAIR

I have created all the templates for the Interns, spend a couple of hours revising the [GitHub Project](https://github.com/orgs/ersilia-os/projects/1) and updating the tasks. I have been working on identifying the bug in the GitActions, solved in this [issue](https://github.com/ersilia-os/eos-template/issues/29).
I have set a meeting with @miquelduranfrigola to discuss in detail his comments on the [Model testing discussion](https://github.com/ersilia-os/ersilia/discussions/700) but I haven't been able to start writing yet, i am to do so by the end of the week.
Looking forward to the WCAIR next lesson!

Pro tip: adding the links to the issues and discussions you mention will be very helpful!

**Comment by pittmanriley** [comment]
Wednesday June 21
- [x ] Get Ersilia up and running (wasn't working yesterday)
- [x ] Familiarize myself more with Docker 
- [ ] Test eos7pw8 https://github.com/ersilia-os/eos7pw8
- [ ] Test eos81ew https://github.com/ersilia-os/eos81ew/issues/8
- [ ] Clean up and dockerize eos7ack https://github.com/ersilia-os/eos7ack/issues/1
- [ ] Test eos74bo https://github.com/ersilia-os/eos74bo/issues

**Comment by GemmaTuron** [comment]
Hi @pittmanriley 

Great, if you could briefly summarize the issue with BentoMl that you identified and how you solved it with Miquel it would be great!

**Comment by pittmanriley** [comment]
Thursday June 22: 

- [x] Get an understanding for why my models won't fetch (such as the model eos74bo)
- [ ] Test the models that I am able to fetch -- need to find out which these are 
- [ ] Clean Up and Dockerize model eos7ack https://github.com/ersilia-os/eos7ack/issues/1

**Comment by pittmanriley** [comment]
Friday June 23: 

- [x] Finish testing eos7ack and submit pull request
- [ ] Start working on presentation for installing for M1 and M2 users

**Comment by GemmaTuron** [comment]
Thanks @pittmanriley
Is there a problem in the fetching of eos96ia? is it due to the mac compatibility issues still?

**Comment by pittmanriley** [comment]
Hi @GemmaTuron 

I’m not sure if the issue is related to mac compatibility. I keep getting an error saying that the model couldn’t produce an output. I tried looking at the troubleshooting webpage, but couldn’t make it work. I will try to work on it further and post an issue if I cannot get it.

**Comment by GemmaTuron** [comment]
Hi @pittmanriley 

To debug a model, you need to:
1. Clone to your local system
2. Use the repo_path flag to try the fetching from local
3. Use the verbose option so you see the whole log, collect it and explain in the relevant issue what have you found in the log files that could point you to the error -- the issue looks empty now: https://github.com/ersilia-os/eos96ia/issues/1

All the above steps are explained in the troubleshooting page in case you need more info. Also remember you can use the slack channel to ask for help if you are unsure about the next steps to take.

**Comment by pittmanriley** [comment]
Tuesday June 27: 

- [x] Meet with the team and with mentors @8-9:30
- [ ] Finish updating model eos96ia, and submit PR 
- [ ] Debug eos7ack 
- [ ] Test eos2r35 
- [ ] Retest eos7pw8 following Gemma's updates to the code
- [ ] If time, test eos8d8a

**Comment by pittmanriley** [comment]
Wednesday June 28: 

- [ ] Attend meeting to go over handling of wrong inputs 
- [x] Finish testing eos8d8a
- [ ] Debug eos2re5
- [ ] Debug eos96ia

**Comment by GemmaTuron** [comment]
Hi @pittmanriley !

Thanks for the summary, good progress. Do not submit a PR until the model works, but do add more information on the issue so I can provide further help! Remember to attach the log files of the errors you get so we can have a more active discussion and you don't get lost!
Let's aim at closing models eos96ia and eos8d8a before moving on

**Comment by pittmanriley** [comment]
Thursday June 29: 

- [ ] Write eos8d8a issue (discussed with Miquel)
- [ ] Troubleshoot eos96ia 
- [ ] Refactor eos8d8a

**Comment by GemmaTuron** [comment]
@pittmanriley thanks, I've requested Simran's help for eos96ia, see if you could summarise to her the main issues!
I've given a few pointers for eos8d8a, which is a complex model, have a look and think how we could structure this repo

**Comment by pittmanriley** [comment]
Sunday July 2: 

- [ ] submit the PR for model eos2mrz

**Comment by pittmanriley** [comment]
Monday July 3: 

- [ ] delete forked repositories that are closed

**Comment by GemmaTuron** [comment]
Hi @pittmanriley 

We'll discuss in the meeting today about the indexing error, for the rest of the tasks all looks good, open PR's when ready and confirm the model testing works. Once these open issues are complete I'll assign new models

**Comment by pittmanriley** [comment]
Tuesday July 4: 

- [ ] attend weekly meeting and internship meeting 
- [ ] submit PR for eos8d8a 
- [ ] resubmit PR for eos2mrz 
- [ ] write necessary issues for eos7ack and try to tackle them
- [ ] test eos7asg 
- [ ] refactor eos2lm8
- [ ] if time, start refactoring eos78ao

**Comment by GemmaTuron** [comment]
Hi @pittmanriley 

Hellen amended model eos7asg so its ready for testing. I won't assign new models unless you let me know you've finished the open ones

**Comment by pittmanriley** [comment]
Wednesday July 5: 

- [ ] clean up space on computer using Gemma's recommendations on Slack 
- [ ] test eos7asg 
- [ ] refactor eos2lm8 
- [ ] refactor eos78ao
- [ ] begin planning out test project

**Comment by GemmaTuron** [comment]
Hi @pittmanriley 

Let's focus on the index errors - we need to see where the models are failing, which can be done with the -v command and adding print statements in the different functions.

**Comment by pittmanriley** [comment]
Thursday July 6: 

- [ ] troubleshoot index issue with eos2lm8
- [ ] test eos8qga
- [ ] refactor eos78ao

**Comment by GemmaTuron** [comment]
Thanks for the update @pittmanriley, let's focus on solving the eos2lm8 and eos78ao by the end of the week if possible. I've answered on Slack as well

**Comment by pittmanriley** [comment]
Friday July 7: 

- [ ] Troubleshoot eos2lm8 and eos78ao 
- [ ] Plan more for test project 

I spent a lot of time troubleshooting eos2lm8 on Friday and over the weekend. I'm sort of stuck on where I should go - I sent a message on Slack about this model, but I'm unable to test the model after making adjustments to the code because I can only run on Codespaces, but when I try to fetch using --repo_path, an error occurs. I tried spending more time on the test project for now. For eos78ao, I'm unable to fetch in CLI and Codespaces for now, so I'm stuck on this as well.

**Comment by miquelduranfrigola** [comment]
Thanks @pittmanriley 

@GemmaTuron - I was looking into eos2lm8. There is an open PR from Ife https://github.com/ersilia-os/eos2lm8/pull/3 (only the README file)

@pittmanriley - let's discuss over Slack and then reflect the relevant points here about eos2lm8. Also, thanks for pushing the "test project" 👌

**Comment by pittmanriley** [comment]
With eos2lm8, I'm still a bit stuck, but I saw that Miquel mentioned there is a pull request pending. For eos78ao, I made some progress because I was able to get it to run by running run.sh in Codespaces, but I'm just not sure if I still need to get it to fetch normally before refactoring (I discuss this in the issues). With eos2mrz, I'm also stuck because the model is working on my system, and it's unclear to me what the error about updating the metadata to Airtable means. Febie and I also made progress on eos7ack today by finding which SMILES seem to cause issues, but we don't find consistent results when we test on this model, so we need Gemma and Miquel's help on this one. Tomorrow, I need to test eos7nno and get started on coding for the test command.

**Comment by pittmanriley** [comment]
Tuesday July 11: 

- [ ] Attend weekly meeting and internship meeting
- [ ] Test eos7nno
- [ ] Troubleshoot eos78ao using Gemma's recommendations
- [ ] Troubleshoot eos2lm8 using Gemma's recommendations 
- [ ] Start coding test command

**Comment by GemmaTuron** [comment]
Hi @pittmanriley 

Good progress, let's finish the open models before focusing on the test command. Please:
- Open a PR for eos2lm8 so we can see if the model test works
- Check the outputs reported by @febielin on model eos7ack

**Comment by pittmanriley** [comment]
Wednesday July 12: 

- [ ] Troubleshoot eos2lm8 and submit PR 
- [ ] Contact SwissADME
- [ ] Test command project 

I tried to get a better understanding for the error happening with eos2lm8, and while I understand the error now, I don't know how to fix it. I talk about this in the issues. For eos7ack, we saw during the meeting on Tuesday that SwissADME blocked some of us, so we need to contact them to get help using their model. I'm unable to go to the SwissADME website and run any SMILES inputs.

**Comment by pittmanriley** [comment]
Thursday July 13: 

- [ ] Submit PR for eos2lm8 
- [ ] Work on test command

**Comment by GemmaTuron** [comment]
Hi @pittmanriley !

I've merged the PR on eos2lm8, let's see if the tests are passed. I won't assign new models because you are onto the test command work, but make sure to complete model testing if any is assigned to you

**Comment by pittmanriley** [comment]
Sounds good!

Friday July 14: 

- [ ] Get test command to run on single SMILES input 
- [ ] Work on basic error handling

**Comment by pittmanriley** [comment]
Monday July 17: 

- [ ] Share my presentations in Google Drive

**Comment by GemmaTuron** [comment]
@pittmanriley 
Please also summarize the work you are doing on the test module, and I see you also need to work on model eos7ack and eos2lm8.
Please prioritize closing off the models before working on the test module, and also run any model testing assigned to you.

**Comment by pittmanriley** [comment]
@GemmaTuron, of course. I believe the PR for eos2lm8 has already been accepted, and I'll respond to the eos7ack issues.

With the test module, I had some trouble last week understanding how to incorporate the run functionality into the ModelTester class. I needed to spend a day or so to get a better understanding of all the classes and how they work together. Over the weekend I spent some more time on it and got it to test with a single output. It works for now, but I would like to go over the organization of the code with you and Miquel sometime this week. As of now, the command just tests a single smiles string, and has very basic error handling, but it works. I've pushed the code to my repository if you'd like to see. For my next steps, I'd like to: 

- add in functionality that sends the output log to a specified output file 
- make it so it tests multiple inputs (instead of just one)

**Comment by pittmanriley** [comment]
Tuesday July 18: 

- [x] Attend weekly meeting and internship meeting 
- [ ] Work on eos96ia 
- [ ] Revisit eos6tg8
- [ ] Refactor eos5smc
- [ ] Add in input/output functionality to the test module

**Comment by pittmanriley** [comment]
Wednesday July 19: 

- [ ] Try to troubleshoot eos5smc 
- [ ] Submit PR for eos96ia
- [ ] Work on to do for test module: 
    - [ ] Make it so outputs of multiple tests all get printed to output file without overwriting 
    - [ ] Work with Febie to test for inconsistency 
    - [ ] Test it with normal run and then try the bash run.sh, comparing the two outputs

**Comment by pittmanriley** [comment]
Thursday July 20: 

- [ ] Test eos5smc using Gemma's recommendations
- [ ] Make adjustments to test module 
    - [ ] Complete the consistency checks, and clean up the code to prevent repetitive code 
    - [ ] Make the output to the console look cleaner and easier to follow for the user 
    - [ ] Work on getting it to run run.sh using bash 
    - [ ] Make it so it checks that the number of outputs equals the number of inputs 
    - [ ] Add in additional error handling for the consistency and previous bullet

**Comment by pittmanriley** [comment]
Friday July 21: 

- [ ] Internship meeting with Miquel
- [ ] Respond to issues about eos5smc and eos96ia
- [ ] Implement recommendations that Miquel has for test module: 
    - [ ] Create the 5% rule to determine if the outputs are consistent. If it over 5%, raise a warning 
    - [ ] Clean up output to the console to make it easier to follow for users 
    - [ ] Implement the run.sh using bash

**Comment by pittmanriley** [comment]
Monday July 24: 

- [ ] Keep looking into eos5smc and eos96ia
- [ ] Clean up the 5% threshold code in the test module 
- [ ] implement run.sh using bash in the test module 

I am not getting anywhere with eos5smc and eos96ia, and hopefully I can get some direction with Miquel tomorrow. I think that the new update to GitHub regarding python 3.11 is affecting the eos96ia PR. For the test module, I got much further today with Febie's help, but we aren't quite finished with the run.sh part yet. We're getting some errors that are taking some time to understand.

**Comment by GemmaTuron** [comment]
Hi @pittmanriley 

I've provided some more info on eos5smc, the error logs are quite clear as to what dependencies are failing. I've also assigned you a new model to work on once you are done with the current tasks.

**Comment by pittmanriley** [comment]
Tuesday July 25: 

- [ ] Solve the dependency issues with eos5smc 
- [ ] Start working on eos9c7k
- [ ] Fix the output issues with the test module 
- [ ] Work on the getting the test module to run using bash run.sh

**Comment by pittmanriley** [comment]
Wednesday July 26: 

- [ ] Finish refactoring eos9c7k 
- [ ] Finish output issues with test module 
- [ ] Keep working on running run.sh within test module

**Comment by pittmanriley** [comment]
Thursday July 27: 

- [ ] Present during team meeting and attend internship meeting 
- [ ] Finish output issues with test module 
- [ ] Begin implementing Miquel's recommendations with consistency checks (using the vector distances)

**Comment by pittmanriley** [comment]
Friday July 28: 

- [ ] Look at the failed PR for eos9c7k 
- [ ] Resubmit PR for eos96ia 
- [ ] Finish output issues with test module 
- [ ] Finish implementing the consistency checks using vector distances

**Comment by pittmanriley** [comment]
Monday July 31: 

- [ ] update the workflows for eos9c7k, eos96ia, and eos5smc 
- [ ] retry fetching eos96ia now that Codespaces is working again 
- [ ] submit PR for eos9c7k, eos96ia, and eos5smc 
- [ ] Continue implementing consistency checks in test module

**Comment by GemmaTuron** [comment]
Hi @pittmanriley 

Great work, thanks! I have merged your PR's. There are a few models assigned to you for testing. I'd say focus on the test module to make it complete, but let's discuss in today's meeting about this!

**Comment by pittmanriley** [comment]
Tuesday August 11: 

- [ ] Test Module:
    - [ ] Change the output functionality to just be a .json file with basic information about how the test went for the model
    - [ ] Consistency check 
- [ ] Test new models: eos6o0z, eos2re5, eos8451, eos4q1a
- [ ] Adjust code for eos9c7k to make the .csv file show each compound as a column 
    - in the service.py file, the output should be .csv instead 
    - use eos7jlv for reference

**Comment by pittmanriley** [comment]
Wednesday August 2: 

- [ ] Keep working on same tasks for test module above
- [ ] Finish testing eos2re5
- [ ] Test eos8451 and eos4q1a
- [ ] Finish fixing the output layout for eos9c7k

**Comment by pittmanriley** [comment]
Thursday August 3: 

- [ ] Finish testing eos4q1a
- [ ] Work on the consistency checks for the test module

**Comment by pittmanriley** [comment]
Monday August 7: 

- [ ] Finish testing eos4q1a 
- [ ] Keep testing the test module on different models looking for bugs
- [ ] Help Febie with running the run.sh with bash in the test module

**Comment by pittmanriley** [comment]
Tuesday August 8: 

- [ ] Retest eos6o0z after the new PR 
- [ ] Mention issue with eos74bo (same issue that was with eos6o0z)
- [ ] Keep testing test module on different models 
- [ ] Document the code and show Gemma
- [ ] Work on running run.sh in the test module

**Comment by GemmaTuron** [comment]
Hi @pittmanriley !

Great job, I've merged the PR. While people try out the current test command, please keep me updated on the run.sh issue, probably it's just a paths thing, Could you show me the specific piece of code for that?

**Comment by pittmanriley** [comment]
Wednesday August 9: 

- [ ] Fix the inconsistent output type issue with eos74bo 
- [ ] Retest eos8451 
- [ ] Look into eos9c7k issue 
- [ ] Keep working on running run.sh with bash in the test module

**Comment by pittmanriley** [comment]
Thursday August 10: 

- [ ] Test eos3ae7
- [ ] Begin working on getting PubChemSampler into Ersilia

**Comment by pittmanriley** [comment]
Friday August 11: 

- [ ] Test eos3ae7 on Colab again and report the results 
- [ ] Keep working on incorporating PubChemSampler

**Comment by GemmaTuron** [comment]
Thanks for the update @pittmanriley 
Let me know how the work on PubChem Sampler is going, where you able to install and run it from chem-sampler?

**Comment by pittmanriley** [comment]
Hi @GemmaTuron, I was able to install and run PubChem. Here's the output I got if you'd like to see: 
[pubchem.log](https://github.com/ersilia-os/ersilia/files/12331495/pubchem.log)

I've been trying to figure out how to incorporate the new model, eos2hzy, from PubChem. What should my steps be? I see that eos2hzy has all the necessary files (main.py, service.py, etc.), but do I need to edit these?

**Comment by GemmaTuron** [comment]
Hi @pittmanriley 

Good that you managed to runt he model. There are step by step instructions on model incorporation in Ersilia's documentation - please follow those: https://ersilia.gitbook.io/ersilia-book/ersilia-model-hub/contribute-models

**Comment by pittmanriley** [comment]
Monday August 14: 

- [ ] Incorporate the PubChem model 
- [ ] Test what Febie has done for the test module 

@GemmaTuron thank you, I forgot those steps were there! I just have to modify the main.py file, which I'll focus on tomorrow.

**Comment by pittmanriley** [comment]
Tuesday August 15: 

- [ ] Finish incorporating the PubChem model

**Comment by pittmanriley** [comment]
Wednesday August 16: 

- [ ] Keep working on the PubChem model 
- [ ] Help Febie test the bash run in the test module

**Comment by GemmaTuron** [comment]
Hi @pittmanriley !
It was great to work with you, thanks so much for your contributions to Ersilia, we hoped you learnt and enjoyed as much as we did! Please, remain engaged with the community and feel free to open any issues or contribute to open ones :)
I'll now close this issue as completed !

**Comment by pittmanriley** [comment]
@GemmaTuron Thank you so much as well! Working with you and the team was a great experience for me this summer. I feel that I learned and grew so much. I'm thankful for this opportunity!
